util/git/client.go

```diff
@@ -208,6 +210,31 @@ func WithEventHandlers(handlers EventHandlers) ClientOpts {
 	}
 }
 
+// WithSparse configures sparse-checkout with the given paths (cone mode)
+// Paths should be directory paths relative to the repository root.
+// Requires Git >= 2.25 for cone mode support.
+func WithSparse(paths []string) ClientOpts {
+	return func(c *nativeGitClient) {
+		if len(paths) > 0 {
+			c.sparsePaths = append([]string{}, paths...)
+		}
+	}
+}
+
+// GetSparseCheckoutKey returns a cache key component for sparse checkout paths.
+// Returns empty string if no sparse paths are configured.
+// The paths are sorted to ensure consistent cache keys regardless of path order.
+func GetSparseCheckoutKey(paths []string) string {
+	if len(paths) == 0 {
+		return ""
+	}
+	// Sort to ensure consistent cache keys regardless of path order
+	sortedPaths := make([]string, len(paths))
+	copy(sortedPaths, paths)
+	sort.Strings(sortedPaths)
+	return "sparse:" + strings.Join(sortedPaths, ",")
+}
```

Comment:
With a lot of paths this cache key could get quite unwieldy. I would prefer for this to be a hash of the sorted paths keys